python/salpytools/salpylib.py

```diff
@@ -157,163 +154,220 @@ def reply_to_transition(self, cmdid):
-        elif self.Stype=='Command':
-            self.myData = getattr(self.SALPY_lib,'{}_command_{}C'.format(self.Device,self.topic))()
-            self.mgr.salProcessor("{}_command_{}".format(self.Device,self.topic))
-            # Generic method to get for example: self.mgr.acceptCommand_takeImages(event)
-            self.acceptCommand = getattr(self.mgr,'acceptCommand_{}'.format(self.topic))
-            self.log.debug("{} subscriber ready for Device:{} topic:{}".format(self.Stype,self.Device,self.topic))
+        Event: It will have (2) underscores. One after the subsytem tag,
+        followed by "logevent", then the second underscore. Ending with an
+        arbitary set of chars for the name of the event.
+
+                         [subsystem_tag]_logevent_[a-zA-Z]
+                                        ex;
+                              scheduler_logevent_target
+
+        Command: Similiar to event, only difference being is that it will have
+        "command" rather than "logevent"
+
+                         [subsystem_tag]_command_[a-zA-Z]
+                                        ex;
+                              scheduler_command_target
+        """
+
+        parsed_topic = topic.split('_')
+
+        if len(parsed_topic[0]) == 0:     # Empty leading string
+            raise ValueError("Please check Topic format")
+
+        if len(parsed_topic[-1]) == 0:    # Empty ending string  
+            raise ValueError("Please check Topic format")
+
+        if len(parsed_topic) < 1:         # No underscores in string
+            raise ValueError("Please check Topic format")
+
+        if len(parsed_topic) > 3:         # Too many underscores in string
+            raise ValueError("Please check Topic format")
+
+        if len(parsed_topic) == 3:        # Middle word not logevent or command
+
+            if parsed_topic[1] == "command" or parsed_topic[1] == "logevent":
+                return parsed_topic[0]
+            else:
+                raise ValueError("Please check Topic format")
+
+        if len(parsed_topic) == 2:
+            return parsed_topic[0]
+
+        raise ValueError("Please check topic format")
+
+    def parse_for_short_topic_name(self, topic):
+
+        parsed_topic = topic.split("_")
+
+        return parsed_topic[-1]
+
+    def configure(self):               # Example Equivilent of...
+        self.set_salpy_lib()           # import SALPY_scheduler
+        self.set_mgr()                 # self.mgr = SAL_scheduler()
+        self.mgr_subscribe_to_topic()  # self.mgr.salEvent("scheduler_[topic]")
+        self.set_data()                # self.data = scheduler_logevent_[topic]C 
+
+    def set_salpy_lib(self):
+        # For example, the equivilent of `import SALPY_scheduler'        
+        try:
+            self.salpy_lib = import_module('SALPY_{}'.format(self.subsystem_tag))
+        except Exception as e:
+            raise e
+
+    def set_mgr(self):
```

Comment:
same as for `set_salpy_lib`...

Reply:
Same comment for set_salpy_lib, test driven development.